tests/test_util_tx.py

```diff
@@ -0,0 +1,119 @@
+import json, logging, pytest
+from decimal import Decimal
+from cryptoadvance.specter.helpers import alias, generate_mnemonic
+from cryptoadvance.specter.key import Key
+from cryptoadvance.specter.process_controller.node_controller import NodeController
+from cryptoadvance.specter.rpc import BitcoinRPC
+from cryptoadvance.specter.specter import Specter
+from cryptoadvance.specter.device import Device
+from cryptoadvance.specter.specter_error import SpecterError
+from cryptoadvance.specter.managers.wallet_manager import WalletManager
+from cryptoadvance.specter.util.tx import convert_rawtransaction_to_psbt
+
+
+def test_import_raw_transaction(
+    caplog,
+    bitcoin_regtest: NodeController,
+    specter_regtest_configured: Specter,
+    hot_wallet_device_1: Device,
+    hot_wallet_device_2: Device,
+):
+
+    caplog.set_level(logging.INFO)
+    devices = [hot_wallet_device_1, hot_wallet_device_2]
+    wallet_manager = specter_regtest_configured.wallet_manager
+
+    def sign_tx_with_device(raw_tx: str, devices=devices) -> str:
+        signed = hot_wallet_device_1.sign_raw_tx(raw_tx, wallet)
+        assert signed["complete"]
+        return signed["hex"]
+
+    def sign_psbt_with_devices(psbt: dict, devices=devices) -> str:
+        signed = psbt
+        for device in devices:
+            signed = device.sign_psbt(signed["psbt"], wallet)
+            if signed["complete"]:
+                break
+        assert signed["complete"]
+        return signed["psbt"]
+
+    # Let's test that with all the relevant key-types
+    for key_type in [
+        "wpkh",  # Single (Segwit)
+        "sh-wpkh",  #  Single (Nested)
+        "wsh",  # Multisig (Segwit)
+        "sh-wsh",  # Multisig (Nested)
+        "tr",  # Taproot
+    ]:
+
+        logging.info(f"begin of loop key_type '{key_type}'")
+        # choose the devices
+        if key_type in ["wsh", "sh-wsh"]:  # multisig
+            keys = [
+                key
+                for device in devices
+                for key in device.keys
+                if key.key_type == key_type
+            ]
+            used_devices = devices
+        else:
+            keys = [key for key in devices[0].keys if key.key_type == key_type]
+            used_devices = [devices[0]]
+
+        assert keys
+
+        # create the wallet for that key-type
+        wallet = wallet_manager.create_wallet(
+            f"bitcoincore_test_wallet_{key_type}",
+            len(used_devices),
+            key_type,
+            keys,
+            used_devices,
+        )
+
+        logging.info(
+            f"created wallet key_type '{key_type}' keys {[key.json for key in keys]}  wallet.account_map {wallet.account_map}"
+        )
+
+        if key_type == "wpkh":
```

Comment:
Do you know why `hot_wallet_device_1.sign_raw_tx(raw_tx, wallet)` works only for this wallet type?

Reply:
It is defined here: https://github.com/cryptoadvance/specter-desktop/blob/e9791068f0fc680397debc7d2176ae0a35ea9957/src/cryptoadvance/specter/devices/bitcoin_core.py#L230
It is a method of the Bitcoin Core Hot Wallet "Device" in Specter. AFAIK, it hasn't been used anywhere else thus far.
I also tried to use it for the Nested Segwit and Multisig Test Wallets but it became too complicated. (At least this is my theory:) For `signrawtransactionwithwallet`(which is the RPC call used by `sign_raw_tx`) to work with those two you need to pass additional information about the redeem and witness scripts in the RPC call. I managed to make it work with bitcoin-cli "manually" for the Single Sig Nested Segwit case (with one ordinary Core wallet with private keys enabled) but Specter uses basically two wallets here, a watch only wallet in Core and an additional Core wallet with private keys (they get passed on with descriptors from a BIP 39 seed). So, somewhere along those lines our test Core hot wallet doesn't have all the info to sign.

I think the test suite is robust enough for this PR, and I would postpone further digging into this matter to a future use case of such tests - where we then perhaps can do it together with the grandmaster @stepansnigirev ;-). Having said this, @relativisticelectron, feel free to try it out.

Reply:
Thanks for the explanation.